service-coordtransform/src/main/java/fi/nls/paikkatietoikkuna/coordtransform/CoordTransWorker.java

```diff
@@ -0,0 +1,130 @@
+package fi.nls.paikkatietoikkuna.coordtransform;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import fi.nls.oskari.annotation.Oskari;
+import fi.nls.oskari.cache.Cache;
+import fi.nls.oskari.cache.CacheManager;
+import fi.nls.oskari.control.ActionException;
+import fi.nls.oskari.control.ActionParamsException;
+import fi.nls.oskari.control.metadata.MetadataFieldHandler;
+import fi.nls.oskari.domain.SelectItem;
+import fi.nls.oskari.service.OskariComponent;
+import fi.nls.oskari.service.OskariComponentManager;
+import fi.nls.oskari.util.IOHelper;
+import org.springframework.web.context.request.async.DeferredResult;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.ForkJoinTask;
+import java.util.concurrent.RejectedExecutionException;
+
+@Oskari
+public class CoordTransWorker extends OskariComponent {
+
+    private ConcurrentHashMap<String, DeferredResult<CoordinatesPayload>> clientResponseMap;
+    private ConcurrentHashMap<String, TransformParams> paramsMap;
+    private ConcurrentHashMap<String, DeferredResult.DeferredResultHandler> handlerMap;
+    private Cache<Object> resultCache;
+    private static final int READ_TIMEOUT_MS = 1000 * 60 * 15;
+    private static final int CONCURRENT_USERS_LIMIT = 100;
+
+    public static final String RESULT_PENDING = "pending";
+
+    public CoordTransWorker() {
+        clientResponseMap = new ConcurrentHashMap<>();
```

Comment:
Is it more or less work to manage 3 different ConcurrentHashMaps? Could it be better to wrap the clientResponse, params and the handler into a single mutable object and only have one map?

Reply:
Sounds reasonable, I'll refactor the code.